### PR TITLE
Document `across` and add multi-branch workflows guide

### PR DESCRIPTION
**Pull request by aoldershaw**
Depends on https://github.com/concourse/examples/pull/4

localhost_3000_multi-branch-workflows html

### CODE REVIEW

File: lit/docs/guides/git.lit [hunk 1]
```diff
@@ -139,3 +139,203 @@
...
+    Our pipeline will have three stages:
+
+    \ordered-list{
+      Run unit tests
+    }{
+      Build and upload a binary to a blobstore (in our case, we'll use
+      \link{Google Cloud Storage}{https://cloud.google.com/storage})
+    }{
+      Trigger a \code{terraform apply} to deploy our app to a staging
+      environment. The \link{Terraform module}{https://github.com/aoldershaw/examples/blob/master/terraform/staging/main.tf}
```
Comment (aoldershaw) [inline lit/docs/guides/git.lit:198]: Reminder to self - a bunch of these links are going to have to change when https://github.com/concourse/examples/pull/4 gets merged

File: lit/docs/guides/git.lit [hunk 2]
```diff
@@ -139,3 +139,230 @@
...
+        type: registry-image
+        source:
+          repository: aoldershaw/git-branches-resource
+
+      resources:
+      - name: feature-branches
+        type: git-branches
+        source:
+          uri: https://github.com/concourse/examples
+          branch_regex: 'feature/(?P<feature>.*)'
```
Comment (taylorsilva) [inline lit/docs/guides/git.lit:243]: What's up with this regex?
image
```suggestion
          branch_regex: 'feature/.*'
```

File: lit/docs/guides/git.lit [hunk 3]
```diff
@@ -139,3 +139,230 @@
...
+    This \reference{resource-types}{\code{resource_type}} emits a new
+    \reference{resource-versions}{resource version} whenever a branch is created or
+    deleted. It also lets us filter the list of branches by a regular expression.
+    In this case, let's assume our feature branches match the regular expression
+    \code{feature/.*}.
+
+    Below is the full pipeline config for this tracker pipeline:
+
+    \link{\code{examples/pipelines/multi-branch/tracker.yml}}{https://github.com/aoldershaw/examples/blob/master/pipelines/multi-branch/tracker.yml}
+    \codeblock{yaml}{{{
```
Comment (taylorsilva) [inline lit/docs/guides/git.lit:231]: any reason for not using `\remote-codeblock{}` here?

File: lit/docs/guides/git.lit [hunk 4]
```diff
@@ -139,3 +139,230 @@
...
+    Concourse will automatically archive any pipelines for branches that get
+    removed. However, Concourse doesn't know that it should destroy Terraform
+    workspaces when a branch is removed. To accomplish this, we can yet again
+    make use of the \link{Terraform resource}{https://github.com/ljfranklin/terraform-resource}
+    to destroy these workspaces. We'll add another job to the tracker pipeline
+    that figures out which workspaces don't belong to an active branch and
+    destroy them.
+
+    \link{\code{examples/pipelines/multi-branch/tracker.yml}}{https://github.com/aoldershaw/examples/blob/master/pipelines/multi-branch/tracker.yml}
+    \codeblock{yaml}{{{
```
Comment (taylorsilva) [inline lit/docs/guides/git.lit:286]: I can see you wanted to omit some stuff here, explains why you didn't use `\remote-codeblock{}` 👍

File: lit/docs/guides/git.lit [hunk 5]
```diff
@@ -139,3 +139,230 @@
...
+        ...
+
+      - name: staging-env
+        type: terraform
+        source:
+          backend_type: gcs
+          backend_config: &terraform_backend_config
+            bucket: concourse-examples
+            prefix: multi-branch/terraform
+            credentials: ((concourse_artifacts_json_key))
```
Comment (taylorsilva) [inline lit/docs/guides/git.lit:310]: nit
```suggestion
            credentials: ((artifacts_json_key))
```

File: lit/docs/jobs/steps.lit [hunk 6]
```diff
@@ -1037,6 +1040,148 @@ by configuring \reference{schema.step.on_failure} or
     }
   }
 
+  \optional-attribute{across}{[across_var]}{
+    Run a wrapped step multiple times with different combinations of variable
```
Comment (taylorsilva) [inline lit/docs/jobs/steps.lit:1044]: I know in the code the substep is wrapped, but in the pipeline yaml where you write `across` it's at the same level as the step it's manipulating.
```suggestion
    Run a step multiple times with different combinations of variable
```

**Review by taylorsilva — CHANGES_REQUESTED** [review]

File: lit/docs/guides/git.lit (hunk 2 above)
Comment (aoldershaw) [reply lit/docs/guides/git.lit:243]: Ah, should probably explain that a bit better. It's a named capture group in a few language's regex implementations (at least Python and Go). https://github.com/aoldershaw/git-branches-resource uses named capture groups to capture a portion of the branch name for use in setting the pipeline (note the `((.:branch.groups.feature))`, which is using this captured value, omitting the leading `feature/`)

File: lit/docs/guides/git.lit (hunk 5 above)
Comment (aoldershaw) [reply lit/docs/guides/git.lit]: Changed to `((gcp_service_account_key))` - probably a bit more obvious

File: lit/docs/guides/git.lit (hunk 1 above)
Comment (taylorsilva) [reply lit/docs/guides/git.lit:198]: Update the links and then this is good to go 👍

**Review by taylorsilva — CHANGES_REQUESTED** [review]

**Comment by aoldershaw** [comment]
@taylorsilva updated

**Review by taylorsilva — APPROVED** [review]

**Pull request merged**